tracker/tracker.go

```diff
@@ -90,6 +94,38 @@ func (t Tracker) FlushEmitter() {
 	t.Emitter.Flush()
 }
 
+func (t *Tracker) waitForEmitter() {
+	for {
+		if !t.Emitter.IsSending() || t.Emitter.SendChannel == nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
1 second seems like quite a long time to wait. Not a huge issue for some applications (i.e. short lived ones) but if I wanted to use this tracker and BlockingFlush then I might want to check more frequently so I can move on sooner. Maybe make this configurable, specify it in milliseconds and inject the wait time in? Then if I want I could choose 100ms or something.

Reply:
Refactored to make it user configurable with ability to supply a milliseconds to sleep between each flush attempt.